pkg/prompb/types.pb.go

```diff
@@ -253,23 +255,101 @@ func (m *Sample) GetTimestamp() int64 {
 	return 0
 }
 
-// TimeSeries represents samples and labels for a single time series.
-type TimeSeries struct {
-	Labels               []Label  `protobuf:"bytes,1,rep,name=labels,proto3" json:"labels"`
-	Samples              []Sample `protobuf:"bytes,2,rep,name=samples,proto3" json:"samples"`
+type Exemplar struct {
+	// Optional, can be empty.
+	Labels []Label `protobuf:"bytes,1,rep,name=labels,proto3" json:"labels"`
+	Value  float64 `protobuf:"fixed64,2,opt,name=value,proto3" json:"value,omitempty"`
+	// timestamp is in ms format, see pkg/timestamp/timestamp.go for
+	// conversion from time.Time to Prometheus timestamp.
+	Timestamp            int64    `protobuf:"varint,3,opt,name=timestamp,proto3" json:"timestamp,omitempty"`
 	XXX_NoUnkeyedLiteral struct{} `json:"-"`
 	XXX_unrecognized     []byte   `json:"-"`
 	XXX_sizecache        int32    `json:"-"`
 }
 
+func (m *Exemplar) Reset()         { *m = Exemplar{} }
```

Comment:
this should be moved to custom.ts.go and we should implement the slice-reuse logic